### ISSUE TITLE
SQL query failed alarm_state_changes - Numeric value out of range

### ISSUE THREAD

**Issue by ltickett** [netxms/netxms#101]
We're seeing this repeatedly in the log file:

```
2023.11.24 15:40:58.509 *E* [db.drv             ] SQL query failed (Query = "INSERT INTO alarm_state_changes (record_id,alarm_id,prev_state,new_state,change_time,prev_state_duration,change_by) VALUES (?,?,?,?,?,?,?)"): [Microsoft][ODBC SQL Server Driver]Numeric value out of range
```

Looks like it's coming from https://github.com/netxms/netxms/blob/1bbfaa95cda368e3a8cd0faed3748cf7cb1d1ea2/src/server/core/alarm.cpp#L636

I can't think of any obvious reason (e.g. we can't have reached MAX(BIGINT) etc.

We're using SQL server- i'm not sure which other information may be helpful in tracking down the issue?

Thanks

**Comment by 2128506** [comment]
One scenario could be m_lastChangeTime being smaller than m_lastStateChangeTime - result will be huge unsigned number (although I cannot see how this could happen). Also, theoretically user ID could contain some garbage value.
I have added additional check for m_lastChangeTime > m_lastStateChangeTime, will see if this will help.
Also, you could enable DB query trace (it will produce lot of logs though) - then we will see actual binding values in log.

**Comment by ltickett** [comment]
Amazing, thanks @2128506 

When will that make it into a release I can install? Or is there anyway I can patch a relevant file?

I'll avoid the query tracing for the moment- but depending on timeline for the patch etc I may give it a try.

**Comment by 2128506** [comment]
We are currently packing version 4.4.4 which includes this change.

**Comment by ltickett** [comment]
That patch has fixed/stopped the error- thanks so much @2128506 💙
